app/controllers/application_controller.rb

```diff
@@ -54,4 +54,14 @@ def require_authenticated_user
       redirect_to main_app.user_omniauth_authorize_path(:shibboleth)
     end
   end
+
+  def authorize_report_access
```

Comment:
Why are you doing this not via CanCanCan? Seems way easier to just say admins can view reports rather than do this custom.